### PR TITLE
Bundle add to canvas

### PR DESCRIPTION
**Pull request by huwshimi**
Added functionality to add a bundle to the canvas.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- search for a bundle
- view the details
- click "Add to canvas"
- the bundle should be added

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3272/
Test PASSed.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -887,11 +887,14 @@ YUI.add('juju-gui', function(Y) {
...
       var state = this.state;
       var utils = views.utils;
-      var cs = this.get('charmstore');
+      var charmstore = this.get('charmstore');
       React.render(
         <components.Charmbrowser
-          charmstoreSearch={cs.search.bind(cs)}
-          getEntity={cs.getEntity.bind(cs)}
+          charmstoreSearch={charmstore.search.bind(charmstore)}
+          importBundleYAML={this.bundleImporter.importBundleYAML.bind(
```
Comment (urosj) [inline jujugui/static/gui/src/app/app.js]: how are errors handled here?

File: jujugui/static/gui/src/app/components/entity-header/entity-header.js [hunk 2]
```diff
@@ -55,7 +55,24 @@ YUI.add('entity-header', function() {
...
     */
     _handleDeployClick: function(e) {
-      this.props.deployService(this.props.entityModel);
+      var entityModel = this.props.entityModel;
+      var entity = entityModel.toEntity();
+      if (entity.type === 'charm') {
+        this.props.deployService(entityModel);
+        this._closeEntityDetails();
+      } else {
+        var id = entity.id.replace('cs:', '');
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/entity-header/entity-header.js]: why do we need to drop cs: with bundles?

File: jujugui/static/gui/src/app/components/entity-header/test-entity-header.js [hunk 3]
```diff
@@ -120,11 +139,15 @@ describe('EntityHeader', function() {
         title="Add to canvas" />);
   });
 
-  it('adds the service when the add button is clicked', function() {
+  it('adds a charm when the add button is clicked', function() {
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/entity-header/test-entity-header.js]: a charm or a bundle here?

**Comment by urosj** [comment]
Not seen a nicer animation of a deploy of a bundle, yet ... sweeeet!

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: Here we're just passing the reference to the `importBundleYAML` method. If need be we can do some error handling where we call the method here: https://github.com/juju/juju-gui/pull/910/files#diff-7cf7695007c40f36e1ca92951612a296R91

File: jujugui/static/gui/src/app/components/entity-header/entity-header.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/entity-header/entity-header.js]: The entity id is something like 'cs:~charmers/django' but when we use that to do a lookup in the api we insert it into the url like this `https://api.jujucharms.com/charmstore/v4/~charmers/django...` hence needing to drop the `cs:` before generating the URL.

File: jujugui/static/gui/src/app/components/entity-header/test-entity-header.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/entity-header/test-entity-header.js]: The is the deploy charm test. The bundle test was added below.

**Comment by urosj** [comment]
LGTM and +1 on QA

File: jujugui/static/gui/src/app/components/entity-header/entity-header.js (hunk 2 above)
Comment (urosj) [inline jujugui/static/gui/src/app/components/entity-header/entity-header.js]: I'd suggest to add some url resolving function in one place and use that in the code. If URLs change, it'll be hard to find all such examples. But not for this PR.

**Comment by hatched** [comment]
:+1: QA :ok:

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**